Pcap++/header/PcapFilter.h

```diff
@@ -286,19 +294,46 @@ namespace pcpp
 		 * @param[in] ipAddress The IPv4 address to build the filter with. If this address is not a valid IPv4 address an error will be
 		 * written to log and parsing this filter will fail
 		 */
-		void setAddr(const std::string& ipAddress) { m_Address = ipAddress; }
+		void setAddr(const std::string& ipAddress) { this->setAddr(IPAddress(ipAddress)); }
+
+		void setAddr(const IPAddress &ipAddress)
+		{
+			if (!ipAddress.isIPv4() && hasMask())
+			{
+				// TODO: What exception type to throw here?
+				throw std::runtime_error("Attempting to set non-IPv4 address while an IPv4 mask is set. "
+										 "Please clear the mask before setting a non-IPv4 address.");
+			}
+
+			m_Address = ipAddress;
+		}
 
 		/**
 		 * Set the IPv4 mask
 		 * @param[in] ipv4Mask The mask to use. Mask should also be in a valid IPv4 format (i.e x.x.x.x), otherwise parsing this filter will fail
 		 */
-		void setMask(const std::string& ipv4Mask) { m_IPv4Mask = ipv4Mask; m_Len = 0; }
+		void setMask(const std::string& ipv4Mask)
+		{
+			if (!m_Address.isIPv4())
+			{
+				// TODO: What exception type to throw here?
+				throw std::runtime_error("Attempting to set an IPv4 mask on non-IPv4 address. "
```

Comment:
I think `runtime_error` is fine. there is a extra space at the end of the message.

Reply:
Extra space is to have a space between this and the next sentence.
Because the way C++ compiler concatenates string literals the code `"This is " "a multipart string literal"` will produce "This is a multipart string literal" after compilation.

The entire error message is:
"Attempting to set an IPv4 mask on non-IPv4 address. Please set an IPv4 address before setting the mask."

Reply:
You can use `std::invalid_argument` here also

Reply:
Fixed bb69dcb